### PR TITLE
ENH: add Shift_JIS mu in _read_edf_header

### PR DESCRIPTION
**Pull request by rcmdnk**
#### What does this implement/fix?
In some EDF files created in Japan, Shift_JIS mu (`\x83\xCA`) is used in the header for the unit.
This change allows to read the unit correctly and assign the correct scale.

### CODE REVIEW

**Review by larsoner — COMMENTED** [review]
LGTM +1 for merge

But before we do, can you add your change to `doc/changes/latest.inc` in the BUG section? If it's your first PR, please use the `:newcontrib:`, put the change at the top of the BUG section, and add your name to `doc/changes/latest.inc`? We want to give you proper credit for the fix in our release notes!

**Review by larsoner — APPROVED** [review]

**Comment by rcmdnk** [comment]
@larsoner 
Thanks for the quick review!

File: doc/changes/latest.inc [hunk 1]
```diff
@@ -113,6 +113,8 @@ Enhancements
 Bugs
 ~~~~
 
+- Add Shift_JIST mu in :func:`mne.io.edf.read_raw_edf` (:gh:`10356` by :newcontrib:`Michiru Kaneda`)
```
Comment (larsoner) [inline doc/changes/latest.inc:116]: One sticking point that [CircleCI caught](https://app.circleci.com/pipelines/github/mne-tools/mne-python/12967/workflows/90af1314-31ba-4cd0-9a5a-8a3ffc0affd8/jobs/41864) (it treats warnings as errors, so if you search "warning" you'll find it):
```
changes/latest.inc:116: WARNING: py:func reference target not found: mne.io.edf.read_raw_edf
```
We have to refer to these things based on where they're documented on the API page, which in this case is the `mne.io` namespace, not `mne.io.edf`:
```suggestion
- Add Shift_JIST mu in :func:`mne.io.read_raw_edf` (:gh:`10356` by :newcontrib:`Michiru Kaneda`)
```

**Comment by larsoner** [comment]
Thanks @rcmdnk !

**Pull request merged**